src/Console/ConsoleOptionParser.php

```diff
@@ -723,7 +723,7 @@ public function parse(array $argv): array
             if ($isBoolean && !isset($params[$name])) {
                 $params[$name] = false;
             }
-            if ($option->isRequired() && !isset($params[$name])) {
+            if ($option->isRequired() && !isset($params[$name]) && empty($params['help'])) {
```

Comment:
Should we be doing a check on 714ish that looks like

```php
if (isset($params['help']) {
    return [$params, $args];
}
```

Reply:
So parsing doesn't determine if an option is missing?

Reply:
I don't think the missing required parameters matter. If `--help` is used we only want to show the command help.

Reply:
Should we not throw the exception at the bottom of this function then? Should that be moved outside? I'm just trying to follow the logic of the check here.

Reply:
You mean with a list of all required options/arguments that are missing?

Reply:
No. I think I misread the `$param` variable to be the option config help, not the parsed help option.

In that case, this is doing what I was asking in a separate comment.

Reply:
So, bail out early it is? @markstory Any reason for not doing it _before_ the arguments check?

Reply:
Before the arguments check is fine by me. Once we have the parse results and help is set we can return early.